### PR TITLE
AT_057 | Entity > Default: testEditRecord

### PR DESCRIPTION
**Pull request by Staschampion**
AT https://trello.com/c/4csU6wlY/931-at155-entity-default-testeditrecord

### CODE REVIEW

**Review by MichaelMicrof — CHANGES_REQUESTED** [review]
Where is your TC? Why your AT wasn't connected with US? Why did you create this method in separate file? Is this duplicate?

File: src/test/java/EntityDefaultEditRecordTest.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import utils.TestUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class EntityDefaultEditRecordTest extends BaseTest {
+    private void createRecord() {
+        findElement(By.xpath("//i[contains(text(), 'create_new_folder')] ")).click();
```
Comment (MichaelMicrof) [inline src/test/java/EntityDefaultEditRecordTest.java:13]: we have special method for this

File: src/test/java/EntityDefaultEditRecordTest.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        findElement(By.id("pa-entity-form-save-btn")).click();
```
Comment (MichaelMicrof) [inline src/test/java/EntityDefaultEditRecordTest.java:14]: the same

File: src/test/java/EntityDefaultEditRecordTest.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    @Test
+    public void testEditRecord() {
+
+        final List<String> oldValuesText = new ArrayList<>();
+        final List<String> newValuesText = new ArrayList<>();
+
+        TestUtils.scrollClick(getDriver(), getDriver().findElement(By.xpath("//p[contains(text(), 'Default')] ")));
```
Comment (MichaelMicrof) [inline src/test/java/EntityDefaultEditRecordTest.java:23]: and this

File: src/test/java/EntityDefaultEditRecordTest.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        Assert.assertEquals(cells.size(), 9);
+
+        List<WebElement> values = findElements(By.xpath("//td[@class='pa-list-table-th']/a"));
+        for (int i = 0; i < values.size(); i++) {
+            oldValuesText.add(values.get(i).getText());
+        }
+        String userText = cells.get(cells.size() - 1).getText();
+        oldValuesText.add(userText);
+
+        findElement(By.xpath("//button/i[text()='menu']")).click();
```
Comment (MichaelMicrof) [inline src/test/java/EntityDefaultEditRecordTest.java:37]: and this

File: src/test/java/EntityDefaultEditRecordTest.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        getWait().until(TestUtils.movingIsFinished(By.xpath("//a[text()='edit']"))).click();
+
+        WebElement stringField = findElement(By.id("string"));
+        stringField.click();
```
Comment (MichaelMicrof) [inline src/test/java/EntityDefaultEditRecordTest.java:41]: why did you need this lines?

File: src/test/java/EntityDefaultEditRecordTest.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        stringField.clear();
+        stringField.sendKeys("Happy New Year");
+
+        TestUtils.jsClick(getDriver(), findElement(By.id("pa-entity-form-save-btn")));
```
Comment (MichaelMicrof) [inline src/test/java/EntityDefaultEditRecordTest.java:45]: we have special method for this

File: src/test/java/EntityDefaultEditRecordTest.java (hunk 2 above)
Comment (Staschampion) [reply src/test/java/EntityDefaultEditRecordTest.java:14]: no need for me

File: src/test/java/EntityDefaultEditRecordTest.java (hunk 1 above)
Comment (Staschampion) [reply src/test/java/EntityDefaultEditRecordTest.java:13]: no need for me

File: src/test/java/EntityDefaultEditRecordTest.java (hunk 5 above)
Comment (Staschampion) [reply src/test/java/EntityDefaultEditRecordTest.java:41]: which one?

**Review by Staschampion — COMMENTED** [review]
as for me there the line absolutely does not interfere there two actions

File: src/test/java/EntityDefaultEditRecordTest.java (hunk 5 above)
Comment (MichaelMicrof) [reply src/test/java/EntityDefaultEditRecordTest.java:41]: why did you use line 41? For what?

Comment (Staschampion) [reply src/test/java/EntityDefaultEditRecordTest.java:41]: I need click on line for make change!

**Comment by MichaelMicrof** [comment]
There isn't work on this PR for more than 2 weeks.

**Comment by MichaelMicrof** [comment]
Stas, I have already explained to you more than once that in order for your PR to pass, you must at least make corrections to your Trello card. What changes are needed - I also wrote to you. It will take 30 seconds, you have been pushing for the third week. If you don't want to follow the rules, it's up to you. When you do everything right, you can get your PR out of the closed ones again. Until then, don't clutter up the list !!!

**Comment by Staschampion** [comment]
Да откройте трелло !!! Там всё есть и не трогайте мой код!!! Не не лезьте ко мне!!!

**Review by ArinaJur — APPROVED** [review]
Approve

**Comment by Staschampion** [comment]
> Where is your TC? Why your AT wasn't connected with US? Why did you create this method in separate file? Is this duplicate?

TC added please check changes

**Review by MichaelMicrof — APPROVED** [review]

**Pull request merged**